TowerOfHanoi/Launcher.java

```diff
@@ -47,16 +51,22 @@ public static void main(String[] args){
             int toTargetArray;
             int valueToMove = 0;
 
-            // VYH: Validate your input over the first 3 chars, ignore anything else after that ~ CAL: done
-            // validate that the actual input was # # (char space char) ~ CAL: done
             try {
-                if (merp.length >= 3) {
-                    // VYH: Sanitize should return an int 0 1 or 2 if a valid character was entered ~ CAL: done
-                    // VYH: Make Sanitize throw an exception if input is not valid ~ CAL: done
-                    // VYH: catch exception here and move forward with next iteration ~ CAL: done
+                if (x.equalsIgnoreCase("undo")){
+                    Integer moveFrom = moveHistory.pop();
+                    Integer moveTo = moveHistory.pop();
+                    if (moveFrom != null && moveTo != null){
+                        System.out.println(moveMe(stacks[moveFrom], stacks[moveTo]));
+                        currentTurn++;
+                    } 
+                } else if (merp.length == 3 && (int)merp[1] ==32) {
```

Comment:
The merp comparison to the value of space is good, however you can save the casting as follows:
```suggestion
                } else if (merp.length == 3 && merp[1] == ' ') {
```
That way you're comparing between `char` types. 

Later we'll go over magic values :)